plugins/modules/aci_vrf_multicast.py

```diff
@@ -0,0 +1,702 @@
+            )
+
+        child_configs.append(source_specific_multicast_config)
+
+        if bootstrap_router:
+            bsr_config = dict(pimBSRPPol=dict(attributes=dict(name=""), children=[dict(pimBSRFilterPol=dict(attributes=dict(name=""), children=[]))]))
+            if bootstrap_router.get("bsr_filter") is not None:
+                set_route_map_config(
+                    existing_config,
+                    bsr_config["pimBSRPPol"]["children"][0]["pimBSRFilterPol"]["children"],
+                    ["pimCtxP", "pimBSRPPol", "pimBSRFilterPol", "rtdmcRsFilterToRtMapPol"],
+                    bootstrap_router.get("bsr_filter"),
+                )
+
+            rp_updates = bootstrap_router.get("rp_updates")
+            if rp_updates is not None:
+                bsr_config["pimBSRPPol"]["attributes"]["ctrl"] = ",".join(sorted(rp_updates))
+
+            child_configs.append(bsr_config)
+
+        if auto_rp:
+            auto_rp_config = dict(pimAutoRPPol=dict(attributes=dict(name=""), children=[dict(pimMAFilterPol=dict(attributes=dict(name=""), children=[]))]))
+
+            if auto_rp.get("ma_filter") is not None:
+                set_route_map_config(
+                    existing_config,
+                    auto_rp_config["pimAutoRPPol"]["children"][0]["pimMAFilterPol"]["children"],
+                    ["pimCtxP", "pimAutoRPPol", "pimMAFilterPol", "rtdmcRsFilterToRtMapPol"],
+                    auto_rp.get("ma_filter"),
+                )
+
+            rp_updates = auto_rp.get("rp_updates")
+            if rp_updates is not None:
+                auto_rp_config["pimAutoRPPol"]["attributes"]["ctrl"] = ",".join(sorted(rp_updates))
+
+            child_configs.append(auto_rp_config)
+
+        mtu = None
+        control_state = None
+        if pim_setting:
+            mtu = pim_setting.get("mtu")
+            control_state = (
+                ",".join(sorted([PIM_SETTING_CONTROL_STATE_MAPPING.get(v) for v in pim_setting.get("control_state")]))
+                if pim_setting.get("control_state") is not None
+                else None
+            )
+
+        aci.payload(
+            aci_class="pimCtxP",
+            class_config=dict(mtu=mtu, ctrl=control_state),
+            child_configs=child_configs,
+        )
+
+        aci.get_diff(aci_class="pimCtxP")
+
+        aci.post_config()
+
+    elif state == "absent":
+        aci.delete_config()
+
+    aci.exit_json()
+
+
+def get_child_from_existing_config(config, class_names):
+    parent = class_names[0]
+    class_names.remove(parent)
+
+    for child in config.get(parent, {}).get("children", []):
+        if len(class_names) == 1 and class_names[0] in child.keys():
+            return child
+        elif child.get(class_names[0], {}).get("children"):
+            return get_child_from_existing_config(child, class_names)
+
+
+def set_route_map_config(existing_config, new_config, class_names, route_map):
```

Comment:
Does this work when updating route_map from one value to another? or is it not applicable here?

Reply:
changed one of the [test](https://github.com/CiscoDevNet/ansible-aci/pull/525/commits/4a6640e44b8cd06668d3517a7b53753765f7d969) to demonstrate it works

Reply:
ok